core/schema/module.go

```diff
@@ -651,6 +657,67 @@ func (s *moduleSchema) currentModuleWorkdirFile(
 	)
 	return inst, err
 }
+func (s *moduleSchema) moduleGenerateClient(
+	ctx context.Context,
+	mod *core.Module,
+	args struct {
+		Generator dagql.String
+		OutputDir dagql.String
+		LocalSdk  dagql.Optional[dagql.Boolean]
+	},
+) (*core.Directory, error) {
+	generator, err := newSDKLoader(s.dag).sdkForModule(
+		ctx,
+		mod.Query,
+		&core.SDKConfig{
+			Source: args.Generator.String(),
+		},
+		mod.Source.Self,
+	)
+	if err != nil {
+		return nil, fmt.Errorf("failed to load generator module %s: %w", args.Generator, err)
+	}
+
+	requiredClientGenerationFiles, err := generator.RequiredClientGenerationFiles(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get required client generation files: %w", err)
+	}
+
+	// Add extra files required to correctly generate the client
+	var source dagql.Instance[*core.ModuleSource]
+	err = s.dag.Select(ctx, mod.Source, &source, dagql.Selector{
+		Field: "withIncludes",
+		Args: []dagql.NamedInput{
+			{
+				Name:  "patterns",
+				Value: dagql.ArrayInput[dagql.String](requiredClientGenerationFiles),
+			},
+		},
+	})
+	if err != nil {
+		return nil, fmt.Errorf("failed to add module source required files: %w", err)
+	}
+
+	// Clone the module and add it to the deps so its binding are also generated
+	mod.Deps = mod.Deps.Append(mod.Clone())
+
+	// HACK: Is there actually a better
+	// Remove the definitions from the module so they does not conflict with its self dependency
+	mod = mod.CloneWithoutDefs()
+
+	generatedClientDir, err := generator.GenerateClient(
+		ctx,
+		source,
+		mod.Deps,
+		args.OutputDir.String(),
+		args.LocalSdk.GetOr(false).Bool(),
+	)
```

Comment:
Is this all necessary? All you need is the schema json file in `generator.GenerateClient`, so why can't you just do something like
```suggestion
	generatedClientDir, err := generator.GenerateClient(
		ctx,
		source,
		mod.Deps.Append(mod),
		args.OutputDir.String(),
		args.LocalSdk.GetOr(false).Bool(),
	)
```

Reply:
> Is this all necessary? All you need is the schema json file in generator.GenerateClient, so why can't you just do something like

Because as explained in the comment, if you simply add the current module as a dep, the typedef registered by the module will conflict with the same typedef installed as dep.
So to avoid that, I clone the mod first to not impact the original state and remove its current loaded typedef so they can be load as dep.

Reply:
But the line `mod = mod.CloneWithoutDefs()` seems like it's a no-op. You are changing the `mod` var but then just passing `mod.Deps`. I don't see how `mod = mod.CloneWithoutDefs()` could possibly be doing anything

Reply:
I pulled this PR down and ran `TestClientGenerator` with the suggested diff above and they all passed